python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,706 @@
+        ), "Invalid number of edges"
+
+        for idx, num_nodes in enumerate(self._data[STR_NUM_NODES_PER_TYPE]):
+            assert (
+                num_nodes > 0
+            ), f"No. of nodes at index: {idx} is not a valid value."
+
+        for idx, num_edges in enumerate(self._data[STR_NUM_EDGES_PER_TYPE]):
+            assert (
+                num_edges > 0
+            ), f"No. of edges at index: {idx} is not a valie value."
+
+        assert len(self._data[STR_NODE_TYPE]) == len(
+            self._data[STR_NUM_NODES_PER_TYPE]
+        ), "No. of node types does not match with No. of nodes per node type."
+        assert len(self._data[STR_EDGE_TYPE]) == len(
+            self._data[STR_NUM_EDGES_PER_TYPE]
+        ), "No. of edge types does not match with No. of edges per edge type."
+
+        assert len(self._ntype_features) <= len(
+            self._data[STR_NUM_NODES_PER_TYPE]
+        ), "Node types with features does not match node type counts."
+        assert len(self._etype_features) <= len(
+            self._data[STR_NUM_EDGES_PER_TYPE]
+        ), "Edge types with features does not match edge type counts."
+
+        # Check for each existing node feature, corresponding node type is valid
+        for key, _ in self._ntype_feature_files.items():
+            ntype = key[0]
+            assert (
+                ntype in self.ntypes
+            ), f"Node Type: {ntype} is not present in the list of node types."
+
+        # Check for each existing edge feature, corresponding edge type is valid
+        for key, _ in self._etype_feature_files.items():
+            etype = key[0]
+            assert (
+                etype in self.etypes
+            ), f"Edge Type: {etype} is not present in the list of edge types"
+
+    ########################## Prrotected Methods ##############################
+
+    def _init_global_eid_offsets(self):
+        """Function to initialize the global_edge_id offsets for all the
+        edge types present in the input graph.
+
+        global_edge_ids, or global_eids, are unique ids assigned to each
+        edge of the input graph. These ids are expected to be unique ids
+        across the input graph.
+
+        The range of global_edge_ids for each edge type in the graph are
+        stored in a dictionary where key-value pairs are etypes and a tuple
+        which indicates the starting and ending value (exlcusive) of the
+        global_edge_ids. For instance,
+            etype1: (a, b) where
+            etype1 is a string describing the edge type
+            and [a, b) is the range of global_edge_ids for the edge type
+            `etype1`.
+        """
+        etypes = self._data[STR_EDGE_TYPE]
+        etype_counts = self._data[STR_NUM_EDGES_PER_TYPE]
+        for count in etype_counts:
+            assert (
+                isinstance(count, int) and count > 0
+            ), "Invalid value for No. of nodes per type"
```

Comment:
This is etype_count, why the error is about No. of nodes per type?

Reply:
Corrected. the typo.